### PR TITLE
Don't reroute while in parking lot

### PR DESCRIPTION
**Pull request by bsudekum**
Ref: https://github.com/mapbox/mapbox-navigation-ios/issues/247

Before rerouting, we check if the user is inside of a parking lot. If so, we do not reroute.

I'm slightly worried about this one as it could produce some negative behavior. I wonder if we should limit this to only when the user is near the end of the route. 

/cc @1ec5 @frederoni

### CODE REVIEW

File: MapboxNavigation/NavigationViewController.swift [hunk 1]
```diff
@@ -293,6 +293,31 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
             return
         }
         
+        // If the user is in a parking lot, don't reroute
+        guard let mapView = mapView else { return }
+        guard let style = mapView.style else { return }
+        
+        let roadLabelLayerIdentifier = "parkingLot"
```
Comment (1ec5) [inline MapboxNavigation/NavigationViewController.swift]: This variable’s name is misleading.

File: MapboxNavigation/NavigationViewController.swift [hunk 2]
```diff
@@ -293,6 +293,31 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
...
+        var streetsSources = style.sources.flatMap { $0 as? MGLVectorSource }.filter { $0.isMapboxStreets}
+        
+        if streetsSources.isEmpty {
+            let source = MGLVectorSource(identifier: "mapboxStreetsv7", configurationURL: URL(string: "mapbox://mapbox.mapbox-streets-v7")!)
+            style.addSource(source)
+            streetsSources.append(source)
```
Comment (1ec5) [inline MapboxNavigation/NavigationViewController.swift]: This logic is duplicated from the street label code. Would it be possible to factor it out into an extension method on MGLStyle?

**Comment by bsudekum** [comment]
This is going to lead to too many edge cases. Closing for now, maybe will revisit.

**Comment by 1ec5** [comment]
Which edge cases are you thinking of?

**Comment by bsudekum** [comment]
Going to take another look at this. Ideally, when someone is beginning to get closer to the edge of the parking lot polygon, we'd start to reroute them. The thinking here is that the user should have informative information whilst exiting the parking lot.

**Comment by ericrwolfe** [comment]
An alternative solution to suppress voice announcements after reroutes: #374

**Comment by bsudekum** [comment]
Should be fixed in https://github.com/mapbox/mapbox-navigation-ios/pull/386

**Pull request closed without merging**